### PR TITLE
SER-3007 | add gcs file backend, handle upload

### PR DESCRIPTION
**Pull request by pwojcik86**
https://wikia-inc.atlassian.net/browse/SER-3007

Adds a GCS file backend. The backend only supports file uploads for now. Do not use on production.

### CODE REVIEW

**Review by mkalinowski-wikia — APPROVED** [review]

File: lib/Wikia/src/Service/Gateway/KubernetesExternalUrlProvider.php [hunk 1]
```diff
@@ -15,7 +15,7 @@ public function getUrl( $serviceName ) {
 		$url = "${wgServicesExternalDomain}${serviceName}";
 
 		if ( $this->logger ) {
-			$this->logger->debug( 'Url provider', [ 'provider_url' => $url ] );
+//			$this->logger->debug( 'Url provider', [ 'provider_url' => $url ] );
```
Comment (mkalinowski-wikia) [inline lib/Wikia/src/Service/Gateway/KubernetesExternalUrlProvider.php]: ?

File: lib/Wikia/src/Service/Gateway/KubernetesUrlProvider.php [hunk 2]
```diff
@@ -40,7 +40,7 @@ public function __construct( string $environment, string $dataCenter ) {
 	public function getUrl( $serviceName ) {
 		$url = "{$this->env}.{$this->dc}.k8s.wikia.net/$serviceName";
 
-		$this->logger->debug( "Url provider", [ 'provider_url' => $url ] );
+//		$this->logger->debug( "Url provider", [ 'provider_url' => $url ] );
```
Comment (mkalinowski-wikia) [inline lib/Wikia/src/Service/Gateway/KubernetesUrlProvider.php]: ?

File: lib/Wikia/src/Service/Swagger/ApiClient.php [hunk 3]
```diff
@@ -64,7 +64,7 @@ public function callApi($resourcePath, $method, $queryParams, $postData, $header
 
 		// keep sampled logging of all requests, but log all server-side errors (HTTP 500+)
 		if ( $this->logSampler->shouldSample() ||  ( $code >= 500 ) ) {
-			$this->$level( $message, $params );
+//			$this->$level( $message, $params );
```
Comment (mkalinowski-wikia) [inline lib/Wikia/src/Service/Swagger/ApiClient.php]: ?

File: lib/composer/composer/ClassLoader.php [hunk 4]
```diff
@@ -377,7 +377,7 @@ private function findFileWithExtension($class, $ext)
             $subPath = $class;
             while (false !== $lastPos = strrpos($subPath, '\\')) {
                 $subPath = substr($subPath, 0, $lastPos);
-                $search = $subPath . '\\';
+                $search = $subPath.'\\';
```
Comment (mkalinowski-wikia) [inline lib/composer/composer/ClassLoader.php:380]: Not sure which formatting is correct and consistent with the rest of the code

**Review by mkalinowski-wikia — APPROVED** [review]
LGTM but some commented code remained

File: lib/composer/composer/ClassLoader.php (hunk 4 above)
Comment (mszabo-wikia) [reply lib/composer/composer/ClassLoader.php:380]: this code is autogenerated by composer, so it shouldn't be edited manually :)

**Review by mszabo-wikia — APPROVED** [review]

**Review by mszabo-wikia — APPROVED** [review]
Let's run `composer dump-autoload` to generate optimized autoload files.

**Review by mkalinowski-wikia — APPROVED** [review]

**Pull request merged**